bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -524,24 +554,67 @@ impl Time {
     /// ```
     #[inline]
     pub fn to_consensus_u32(self) -> u32 { self.0 }
-}
 
-impl_parse_str_from_int_fallible!(Time, u32, from_consensus, Error);
+    fn from_consensus_i64(n: i64) -> Result<Self, ParseError> {
+        let n = u32::try_from(n).map_err(|_| ParseError::Conversion(n))?;
+        Self::from_consensus(n).map_err(Into::into)
+    }
+}
```

Comment:
In 670c16b912e757ab6e657b65394a5a84505ac42e:

There are two error paths here which both wind up returning the `ParseError::Conversion` variant. Is there a way this function could return `ConversionError` instead?

Reply:
Oh, this whole method is not needed, I messed up hex parsing for `Time`.

But to give a bit of context, it was originally intentional since it's a private method that is only used as parsing primitive and it helps to return the internal type because in theory it makes constructing the other errors simpler. This worked well in my other projects so I just reused the pattern however it turns out it's not really that useful in this code.